lib/mayaUsd/ufe/wrapUtils.cpp

```diff
@@ -81,36 +88,103 @@ std::string stagePath(UsdStageWeakPtr stage)
+    const SdfPath& usdPath,
+    int            instanceIndex = UsdImagingDelegate::ALL_INSTANCES)
 {
-#ifdef UFE_V2_FEATURES_AVAILABLE
-    return ufe::ufePathToPrim(Ufe::PathString::path(ufePathString));
-#else
+    return ufe::usdPathToUfePathSegment(usdPath, instanceIndex).string();
+}
+
+#ifndef UFE_V2_FEATURES_AVAILABLE
+// Helper function for UFE versions before version 2 for converting a path
+// string to a UFE path.
+static Ufe::Path _UfeV1StringToUsdPath(const std::string& ufePathString)
+{
+    Ufe::Path path;
+
     // The path string is a list of segment strings separated by ',' comma
     // separator.
     auto segmentStrings = TfStringTokenize(ufePathString, ",");
 
-    // If there's just one segment, it's the Maya Dag path segment, so it can't
-    // have a prim.
-    if (segmentStrings.size() == 1) {
-        return UsdPrim();
+    // If there are fewer than two segments, there cannot be a USD segment, so
+    // return an invalid path.
+    if (segmentStrings.size() < 2u) {
+        return path;
     }
 
-    // We have the path string split into segments.  Build up the Ufe::Path one
-    // segment at a time.  The path segment separator is the first character
-    // of each segment.  We know that USD's separator is '/' and Maya's
+    // We have the path string split into segments. Build up the Ufe::Path one
+    // segment at a time. The path segment separator is the first character
+    // of each segment. We know that USD's separator is '/' and Maya's
     // separator is '|', so use a map to get the corresponding UFE run-time ID.
-    Ufe::Path                        path;
-    static std::map<char, Ufe::Rtid> sepToRtid = { { '/', ufe::getUsdRunTimeId() }, { '|', 1 } };
-    for (std::size_t i = 0; i < segmentStrings.size(); ++i) {
+    static std::map<char, Ufe::Rtid> sepToRtid
+        = { { '/', ufe::getUsdRunTimeId() }, { '|', ufe::getMayaRunTimeId() } };
+    for (size_t i = 0u; i < segmentStrings.size(); ++i) {
         const auto& segmentString = segmentStrings[i];
-        char        sep = segmentString[0];
+        char        sep = segmentString[0u];
         path = path + Ufe::PathSegment(segmentString, sepToRtid.at(sep), sep);
     }
+
+    return path;
+}
+#endif
+
+std::string stripInstanceIndexFromUfePath(const std::string& ufePathString)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    const Ufe::Path path = Ufe::PathString::path(ufePathString);
+#else
+    const Ufe::Path path = _UfeV1StringToUsdPath(ufePathString);
+#endif
+
+    return ufe::stripInstanceIndexFromUfePath(path).string();
+}
+
+UsdPrim ufePathToPrim(const std::string& ufePathString)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    return ufe::ufePathToPrim(Ufe::PathString::path(ufePathString));
+#else
+    Ufe::Path       path = _UfeV1StringToUsdPath(ufePathString);
+
+    // If there are fewer than two segments, there cannot be a USD segment, so
+    // return an invalid UsdPrim.
+    const size_t numSegments =
+#if UFE_PREVIEW_VERSION_NUM >= 2006
```

Comment:
Because this code path is UFE v1 only, you should be able to get rid of this UFE_PREVIEW_VERSION_NUM pre-processor conditional.

Reply:
Addressed in 34329b23f94983b0fb48c1104f197b99a36caf24.